src/Internal/Contract/WaitUntilSlot.purs

```diff
@@ -50,20 +51,39 @@ waitUntilSlot futureSlot = do
       | slot >= futureSlot -> pure tip
       | otherwise -> do
           { systemStart } <- asks _.ledgerConstants
-          eraSummaries <- liftAff $
-            queryHandle.getEraSummaries
-              >>= either (liftEffect <<< throw <<< show) pure
-          slotLengthMs <- map getSlotLength $ liftEither
+          let
+            getEraSummaries =
+              liftAff queryHandle.getEraSummaries
+                >>= liftContractE
+          eraSummaries <- getEraSummaries
+          highestSlot <- liftContractM "Can't find any Era summary" $
+            highestEndSlotInEraSummaries eraSummaries
+          let
+            toWait = case highestSlot of
+              Just highestEndSlot
+                | Just waitThen <- futureSlot `sub` highestEndSlot
```

Comment:
This subtraction is not needed, because futureSlot is the target slot we wait for. It can't be changed.